src/Chronos.php

```diff
@@ -193,22 +193,12 @@ public static function hasTestNow(): bool
      */
     public function __debugInfo(): array
     {
-        // Conditionally add properties if state exists to avoid
-        // errors when using a debugger.
-        $vars = get_object_vars($this);
```

Comment:
How will the new code work when paused in a debugger? The conditional property building was to work around debuggers needing to format Chronos objects before the constructor was finished. See #188

Reply:
How does I18n\Time work?

https://github.com/cakephp/cakephp/blob/d509a3a820582087c4b3afa8670d00dd600dd191/src/I18n/DateFormatTrait.php#L448

Reply:
Let me give a more useful comment :).

**7.3**
```
php > print_r(get_object_vars(new \DateTime('now')));
Array
(
    [date] => 2019-10-31 03:10:01.434797
    [timezone_type] => 3
    [timezone] => UTC
)
```

**7.4**
```
php > print_r(get_object_vars(new \DateTime('now')));
Array
(
)
```

Reply:
We could get really tricky and have a property that tracks whether the parent constructor was called. This would need to exist in Chronos and CakePHP.

Reply:
@markstory Since Xdebug 2.7.2 won't use `__debugInfo()` for user-defined classes, Xdebug will show real properties.

Reply:
That is a real pity :( Should we open a issue on xdebug that `__debugInfo()` no longer works?

Reply:
`serialize($this)` equals to `O:20:"Cake\Chronos\Chronos":0:{}` if not object initiated.

Check if it ends with `:0:{}`.

Reply:
@dakota 
https://github.com/xdebug/xdebug/pull/469

Reply:
Maybe the question we should be asking is does Date really need __debuginfo().

Reply:
Well if xdebug is no longer calling `__debugInfo()` then we won't have the old problem and these changes will be 👌

Reply:
Looks like Xdebug 2.8 is the first release that supports php 7.4.